src/google/credential.rs

```diff
@@ -27,21 +28,38 @@ pub enum CredentialType {
 }
 
 /// A Google API credential file.
-#[derive(Clone, serde::Deserialize)]
+#[derive(Clone, Default)]
 #[cfg_attr(test, derive(Debug))]
 pub struct Credential {
-    /// The type of the credential file.
-    #[serde(rename = "type")]
-    pub ty: CredentialType,
-
-    #[serde(flatten)]
     pub(crate) service_account: Option<ServiceAccount>,
-    #[serde(flatten)]
     pub(crate) impersonated_service_account: Option<ImpersonatedServiceAccount>,
-    #[serde(flatten)]
     pub(crate) external_account: Option<ExternalAccount>,
 }
 
+impl Credential {
+    /// Deserialize credential file
+    pub fn from_slice(v: &[u8]) -> Result<Credential> {
+        let service_account = serde_json::from_slice(v).ok();
```

Comment:
Hi, we can implement in this way: https://serde.rs/enum-representations.html

Convert them to enum and using `#[serde(tag = "type")]`.

Reply:
Oh, you have tried this.

Reply:
Yeah, I tried:

```rust
#[serde(tag = "type")]
pub enum Credential {
    ImpersonatedServiceAccount,
    ExternalAccount,
    ServiceAccount,
}
```

but then when you deserialize `Credential`, you get:

![image](https://github.com/Xuanwo/reqsign/assets/82846066/8f42dfb5-469a-4ad3-91ac-c9bce28b5788)

and then I couldn't find a way to get a `ServiceAccount` or `ExternalAccount` or `ImpersonatedServiceAccount`, because the type is just `Credential`

Before, you could just write `.service_account` to get a `ServiceAccount`, like:

![image](https://github.com/Xuanwo/reqsign/assets/82846066/d98215b8-a386-4ea4-9d56-19af97dda0ac)

but it doesn't work when deserializing an enum, and I can't find/google/figure out the syntax for how to turn an enum into one of its variants.